mantidproject/mantid
# Improve beam stop masking in DirectILLDiagnostics

This PR extends the hard mask of IN5 spectrometer at ILL to include the direct beam area and fixes the beam stop heuristics in `DirectILLDiagnostics` to take the hard mask into account.

Fixes #21470 

**To test:**

Here is some IN5 test data (the beam stop masking is OK for the IN5 file in Mantid's unit test data):
[161397.nxs.zip](https://github.com/mantidproject/mantid/files/1792615/161397.nxs.zip)

You can try to run the following script:
```python
Load(Filename='161397', OutputWorkspace='161397')
ExtractMonitors(InputWorkspace='161397', DetectorWorkspace='det')
DirectILLDiagnostics(InputWorkspace='det', OutputWorkspace='mask')
MaskDetectors(Workspace='161397', MaskedWorkspace='mask')
```

Open the instrument view for `161397` and observe the diamond-shaped mask covering the entire beam stop area. No tubes should be left unmasked.

**Release Notes** 

The changes are mentioned in the Direct Inelastic release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
